File: tests/fixtures/sample_kotlin/Sample.kt
/**
 * Module-level KDoc.
 */
package com.example

/**
 * A public class with KDoc.
 */
class SampleClass<T>(val publicProp: String, private val privateProp: Int) {
    /**
     * A public method.
     */
    fun publicMethod(input: String = "default"): String {
        return input
    }

    protected fun protectedMethod() {}

    private fun privateMethod() {}

    internal fun internalMethod() {}

    companion object {
        /**
         * Method in companion object.
         */
        fun companionMethod() {}

        val companionProp = 42
    }
}

/**
 * Data class.
 */
data class SampleData(val x: Int, val y: Int)

/**
 * Extension function.
 */
fun String.shout(): String = this.uppercase()

/**
 * Top-level property.
 */
val topLevelVal = "Hello"

private val privateTopLevelVal = "Secret"

/**
 * Sealed class and subclasses.
 */
sealed class Expr
data class Const(val number: Double) : Expr()
data class Sum(val left: Expr, val right: Expr) : Expr()

/**
 * Interface.
 */
interface SampleInterface {
    fun foo()
}

/**
 * Enum.
 */
enum class Color { RED, GREEN, BLUE }

/**
 * Singleton object.
 */
object Singleton {
    fun greet() = "Hello"
}

@Deprecated("use bar")
fun annotatedFun() {}

File: tests/fixtures/sample_php/Sample.php
<?php
/**
 * File-level PHPDoc.
 */

namespace Sample;

/**
 * A public class with PHPDoc.
 */
class SampleClass extends BaseClass implements SampleInterface {
    /**
     * A public constant.
     */
    public const PUBLIC_CONST = 'value';

    /**
     * A public property.
     */
    public string $publicProp = 'default';

    /**
     * A protected property.
     */
    protected $protectedProp;

    /**
     * A private property.
     */
    private $privateProp;

    /**
     * A public method.
     * @param string $input
     * @return string
     */
    public function publicMethod(string $input): string {
        return $input;
    }

    /**
     * An unscoped method (defaults to public).
     */
    function unscopedMethod() {}

    /**
     * A protected method.
     */
    protected function protectedMethod() {}

    /**
     * A private method.
     */
    private function privateMethod() {}
}

/**
 * A global function.
 */
function globalFunction(int $x, $y = 10): void {}

/**
 * An interface.
 */
interface SampleInterface {
    public function foo();
}

/**
 * A trait.
 */
trait SampleTrait {
    public function traitMethod() {}
}

/**
 * An enum.
 */
enum SampleEnum: string {
    case A = 'a';
    case B = 'b';
}
